app/deniedrefs/DeniedRefsRepo.scala

```diff
@@ -46,13 +47,21 @@ final class DeniedRefsRepo @Inject() (
     replaceIndexes = true
   ) {
 
-  //TODO:x bring back the projection attribute so we don't fetch entire collection each time to
-  // see what is the latest it
-  def findLatestDeniedRefsId(): Future[Option[DeniedRefsId]] = collection
-    .find()
-    .sort(Sorts.descending(inserted))
-    .headOption()
-    .map(_.map(_._id))
+  def findLatestDeniedRefsId(): Future[Option[DeniedRefsId]] = findLatestDeniedRefs().map(_.map(_._id))
+
+  /**
+   * Projection is used (i.e. slice("_id", 1) ) to limit the number of records returned to just one.
+   * Projection is also used (i.e. slice("refs", 1) ) to limit the number of refs returned to just one.
+   * We don't need them and it can introduce performance issue if there are lots in list of refs inside DeniedRefs
+   * Don't remove this... unless you know what you're doing ;)
+   */
```

Comment:
I have gone with this instead of jsobject. I ran into issues with existing tests failing when using the jsobject approach (JsResult exceptions due to empty string)
I think we should stick with keeping the "refs" projection in, as it means we don't return a list of 450+ refs in the Seq - we don't use them here, only the _id field is used and this does that, while allowing us to not tweak the repo elsewhere